modules/siddhi-core/src/main/java/org/wso2/siddhi/core/query/processor/stream/window/ExternalTimeBatchWindowProcessor.java

```diff
@@ -82,6 +110,15 @@ protected void init(ExpressionExecutor[] attributeExpressionExecutors, Execution
      */
     @Override
     protected synchronized void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor, StreamEventCloner streamEventCloner) {
+        if(numberOfPrameters ==2){
+            processFor2Prameters(streamEventChunk,nextProcessor,streamEventCloner);
```

Comment:
Minudika, we can have a boolean flag (isStartTimeEnabled or something) and calls corresponding method at that time.. No need an "else if" as well.